bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/handler/FreeDeviceHandler.java

```diff
@@ -0,0 +1,145 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.freeboxos.internal.handler;
+
+import static org.openhab.binding.freeboxos.internal.FreeboxOsBindingConstants.*;
+
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.freeboxos.internal.api.FreeboxException;
+import org.openhab.binding.freeboxos.internal.api.system.DeviceConfig;
+import org.openhab.binding.freeboxos.internal.api.system.Sensor;
+import org.openhab.binding.freeboxos.internal.api.system.Sensor.SensorKind;
+import org.openhab.core.library.CoreItemFactory;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public abstract class FreeDeviceHandler extends HostHandler {
+    private final Logger logger = LoggerFactory.getLogger(FreeDeviceHandler.class);
+    private long uptime = -1;
+
+    public FreeDeviceHandler(Thing thing, ZoneId zoneId) {
+        super(thing, zoneId);
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        try {
+            while (!checkBridgeHandler()) {
+                Thread.sleep(1000);
+            }
+            initializeChannels();
+        } catch (InterruptedException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    @Override
+    protected void internalPoll() throws FreeboxException {
+        super.internalPoll();
+        fetchSystemConfig();
+    }
+
+    protected abstract DeviceConfig getDeviceConfig() throws FreeboxException;
+
+    protected abstract void internalCallReboot() throws FreeboxException;
+
+    private void initializeChannels() {
+        List<Channel> channels = new ArrayList<>(getThing().getChannels());
+
+        try {
+            DeviceConfig systemConfig = getDeviceConfig();
+            List<Sensor> sensors = systemConfig.getAllSensors();
+            sensors.forEach(sensor -> {
+                ChannelUID sensorId = new ChannelUID(thing.getUID(), PROPERTY_SENSORS, sensor.getId());
+                if (channels.stream().noneMatch(c -> c.getUID().equals(sensorId))) {
+                    ChannelBuilder channelBuilder = ChannelBuilder.create(sensorId).withLabel(sensor.getName());
```

Comment:
API is returning not ideal sensor names.
Would you be ok to add either "Vitesse " or "Température " before sensor.getName() in case sensor.getName() does not start by that ?
This will allow to have "Vitesse Ventilateur 1" rather than "Ventilateur 1" and "Température Disque dur" rather than "Disque dur".